File: PHP/Film.php
<?php
// Class Film untuk menyimpan data film di bioskop
class Film {
    // Atribut dibuat private untuk menerapkan enkapsulasi
    private $id;
    private $judul;
    private $genre;
    private $durasi;
    private $gambar;

    // Constructor
    public function __construct($id = 0, $judul = "", $genre = "", $durasi = 0, $gambar = "") {
        $this->id = (int)$id;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasi = (int)$durasi;
        $this->gambar = $gambar;
    }

    // Getter dan Setter
    public function getId() {
        return $this->id;
    }

    public function setId($id) {
        $this->id = (int)$id;
    }

    public function getJudul() {
        return $this->judul;
    }

    public function setJudul($judul) {
        $this->judul = $judul;
    }

    public function getGenre() {
        return $this->genre;
    }

    public function setGenre($genre) {
        $this->genre = $genre;
    }

    public function getDurasi() {
        return $this->durasi;
    }

    public function setDurasi($durasi) {
        $this->durasi = (int)$durasi;
    }

    public function getGambar() {
        return $this->gambar;
    }

    public function setGambar($gambar) {
        $this->gambar = $gambar;
    }
}
?>

File: CPP/Film.cpp
#include <iostream>
#include <string>
using namespace std;

// Class Film digunakan untuk menyimpan data film
class Film
{
private:
    // Atribut dibuat private untuk menerapkan encapsulation
    int id;
    string judul;
    string genre;
    int durasi;
    string gambar;

    public:
        Film(){
        }
        // Mengambil nilai ID
        int getId(){
            return id;
        }
        // Mengambil nilai judul
        string getJudul(){
            return judul;
        }
        // Mengambil nilai genre
        string getGenre(){
            return genre;
        }
        // Mengambil nilai durasi
        int getDurasi(){
            return durasi;
        }
        // Mengambil nilai path gambar
        string getGambar(){
            return gambar;
        }

        // Mengubah nilai ID
        void setId(int idBaru){
            id = idBaru;
        }
        // Mengubah nilai judul
        void setJudul(string judulBaru){
            judul = judulBaru;
        }
        // Mengubah nilai genre
        void setGenre(string genreBaru){
            genre = genreBaru;
        }
        // Mengubah nilai durasi
        void setDurasi(int durasiBaru){
            durasi = durasiBaru;
        }
        // Mengubah nilai path gambar
        void setGambar(string gambarBaru){
            gambar = gambarBaru;
        }
        // Menampilkan seluruh data dari sebuah object Film
        void tampilkanData(){
            cout << "ID       : " << getId() << endl;
            cout << "Judul    : " << getJudul() << endl;
            cout << "Genre    : " << getGenre() << endl;
            cout << "Durasi   : " << getDurasi() << " menit" << endl;
            cout << "Gambar   : " << getGambar() << endl;
        }

        ~Film(){
        }
};
